### ISSUE TITLE
Can't start when no windows are active

### ISSUE THREAD

**Issue by ging-dev** [leukipp/cortile#18]
I tried adding cortile to autostart list in xfce but it seems cortile can't start when no windows are active.
`cortile.log`:
`Error on initialization GetProperty: No such property '_NET_ACTIVE_WINDOW' on window 765.`

**Comment by leukipp** [comment]
I made a potential fix, please try to follow the [dev install instructions](https://github.com/leukipp/cortile/tree/main#development-) and install cortile via remote source from the `develop` branch. Let me know if this version solves your problem.

If not, you may temporarily delay autostart. e.g. via,
```bash
bash -c "sleep 10 && /home/USER/.go/bin/cortile -vvv"
``` 
we can than investigate the issue further.

**Comment by ging-dev** [comment]
> I made a potential fix, please try to follow the [dev install instructions](https://github.com/leukipp/cortile/tree/main#development-) and install cortile via remote source from the `develop` branch. Let me know if this version solves your problem.
> 
> If not, you may temporarily delay autostart. e.g. via,
> 
> ```shell
> bash -c "sleep 10 && /home/USER/.go/bin/cortile -vvv"
> ```
> 
> we can than investigate the issue further.

Thank you, `develop` branch solved the problem!

**Comment by leukipp** [comment]
> Updated: Well it seems it doesn't work sometimes

Can you confirm that the `sleep` command,
```bash
bash -c "sleep 10 && /home/USER/.go/bin/cortile -vvv"
``` 
solves the issue on startup, regardless if windows are active or not in the first 10 seconds?

Please also provide additional infos about the Desktop Environment/Window Manager you are using.

**Comment by ging-dev** [comment]
> > Updated: Well it seems it doesn't work sometimes
> 
> Can you confirm that the `sleep` command,
> 
> ```shell
> bash -c "sleep 10 && /home/USER/.go/bin/cortile -vvv"
> ```
> 
> solves the issue on startup, regardless if windows are active or not in the first 10 seconds?
> 
> Please also provide additional infos about the Desktop Environment/Window Manager you are using.

To make `cortile` start normally, I have to open any window before `cortile` start, if no window is active, `cortile` will fail to start. Using the `sleep` command only works if before 10 seconds a certain window is opened.

i use XFCE desktop environment and Metacity as window manager.

**Comment by leukipp** [comment]
I understand, but unfortunately I can't reproduce the problem properly on my system.

Are you using a desktop manager (e.g. xfdesktop) or are panels (e.g. xfce4-panel) configured, or is the desktop initially completely empty (no background image etc.) before other windows are opened?

Please also run the latest `develop` branch using `cortile -v`, as I have changed some logging levels and let me know what the result is.

**Comment by ging-dev** [comment]
> I understand, but unfortunately I can't reproduce the problem properly on my system.
> 
> Are you using a desktop manager (e.g. xfdesktop) or are panels (e.g. xfce4-panel) configured, or is the desktop initially completely empty (no background image etc.) before other windows are opened?

Yes, my `xfdesktop` and `xfce4-panel` always start up immediately.

> Please also run the latest `develop` branch using `cortile -v`, as I have changed some logging levels and let me know what the result is.

Thank you, looks like this [commit](https://github.com/leukipp/cortile/commit/2f5e196090a3b5cf75c67d6b7ba0192acfb8b25c) fixed the problem, i was able to start `cortile` without the `sleep` command and open the window manually. 👉👈

**Comment by leukipp** [comment]
There might be some new issues related to system startup caused by this commit (346a89b). If you are still using cortile, it would be helpful if you could confirm that version [v2.3.0](https://github.com/leukipp/cortile/releases/tag/v2.3.0) is working as expected.
